pkg/resource/kubeconfig_factory.go

```diff
@@ -23,20 +26,42 @@ func (r *Helper) getKubeconfigFactory(namespace string) (cmdutil.Factory, error)
 	}
 	r.logger.Debug("creating client config from kubeconfig")
 	clientConfig := clientcmd.NewNonInteractiveClientConfig(*config, "", overrides, nil)
+	if r.metricsEnabled {
+		restConfig, err := clientConfig.ClientConfig()
```

Comment:
@dgoodwin the issue is that every time you call ClientConfig you get a new copy of the config. Modifying the local copy won't affect the result of future calls to ClientConfig (as happens in line 62). If you want the same config to be returned every time, simply add a field to kubeconfigClientGetter to set a modified config, and return *that* config in the ToRESTConfig method. The other consideration is that most likely the client config you get will already have a WrapTransport function set. So you will need to wrap *that* function with your function.